### PR TITLE
Adri

### PR DESCRIPTION
**Pull request by amolero6**
2nd try having rebased with master branch... 
Class for dataset treatment that includes building vocabulary based on word frequency from input (pending to use torchtext and spacy).
Dataloader successfully loads each input (image converted to tensor through PIL) and target (caption converted to vector with vocabulary)

### CODE REVIEW

File: dataset.py [hunk 1]
```diff
@@ -1,77 +1,167 @@
...
+import nltk
 import pandas as pd
+import torch
 from torch.utils.data import Dataset, DataLoader
+from torch.nn.utils.rnn import pad_sequence
 from PIL import Image
 from torchvision import transforms
+
+
+class Vocabulary():
```
Comment (lucaswerner90) [inline dataset.py:11]: put the class in a different file so is much clear to identify where it's. In general I would suggest a class per file

File: dataset.py [hunk 2]
```diff
@@ -1,77 +1,167 @@
...
+            image = self.transform(image)
 
-		return image, caption
+        tok_vec = []
+        tok_vec += [self.vocab.word_to_index['<START>']]
+        tok_vec += self.vocab.numericalize(caption)
+        tok_vec += [self.vocab.word_to_index['<END>']]
 
+        return image, torch.tensor(tok_vec)
+class CapsCollate:
```
Comment (lucaswerner90) [inline dataset.py:125]: same for this one. Maybe a module should be created in the project that contains the whole dataset classes structure

File: .gitignore [hunk 3]
```diff
@@ -128,3 +128,5 @@ dmypy.json
 
 # Pyre type checker
 .pyre/
+#Custom
+notes.txt
```
Comment (lucaswerner90) [inline .gitignore:132]: Even though you added it notes.txt here, it seems that it was added to the pull request anyway. Just deleted it from it

**Review by lucaswerner90 — CHANGES_REQUESTED** [review]
Minor comments that can be addressed later. Feel free to merge it

**Comment by amolero6** [comment]
Added a file per class and tested again

**Review by lucaswerner90 — APPROVED** [review]
looks fine

**Pull request merged**